tsdb/engine/tsm1/compact_test.go

```diff
@@ -2269,837 +2270,510 @@ func TestDefaultPlanner_PlanOptimize_NoLevel4(t *testing.T) {
+			}, []int{}, "not fully compacted and not idle because of more than one generation", 3,
+		},
+		{
+			// This test will mock a 'backfill' condition where we have a single
+			// shard with many generations. The initial generation should be fully
+			// compacted, but we have some new generations that are not. We need to ensure
+			// the optimize planner will pick these up and compact everything together.
+			"Backfill mock condition",
+			[]tsm1.FileStat{
+				{
+					Path: "01-05.tsm1",
+					Size: 2048 * 1024 * 1024,
+				},
+				{
+					Path: "01-06.tsm1",
+					Size: 2048 * 1024 * 1024,
+				},
+				{
+					Path: "01-07.tsm1",
+					Size: 2048 * 1024 * 1024,
+				},
+				{
+					Path: "02-04.tsm1",
+					Size: 700 * 1024 * 1024,
+				},
+				{
+					Path: "02-05.tsm1",
+					Size: 500 * 1024 * 1024,
+				},
+				{
+					Path: "02-06.tsm1",
+					Size: 400 * 1024 * 1024,
+				},
+				{
+					Path: "03-02.tsm1",
+					Size: 700 * 1024 * 1024,
+				},
+				{
+					Path: "03-03.tsm1",
+					Size: 500 * 1024 * 1024,
+				},
+				{
+					Path: "03-04.tsm1",
+					Size: 400 * 1024 * 1024,
+				},
+				{
+					Path: "04-01.tsm1",
+					Size: 700 * 1024 * 1024,
+				},
+				{
+					Path: "04-02.tsm1",
+					Size: 500 * 1024 * 1024,
+				},
+				{
+					Path: "03-03.tsm1",
+					Size: 400 * 1024 * 1024,
+				},
+			}, []int{
+			tsdb.AggressiveMaxPointsPerBlock,
+			tsdb.AggressiveMaxPointsPerBlock,
+			tsdb.AggressiveMaxPointsPerBlock,
+
+			tsdb.AggressiveMaxPointsPerBlock,
+			tsdb.AggressiveMaxPointsPerBlock,
+			tsdb.DefaultMaxPointsPerBlock,
+
+			tsdb.AggressiveMaxPointsPerBlock,
+			tsdb.AggressiveMaxPointsPerBlock,
+			tsdb.DefaultMaxPointsPerBlock,
+
+			tsdb.DefaultMaxPointsPerBlock,
+			// Use some magic numbers but these are just small values for block counts
+			100,
+			10,
+		},
+			"not fully compacted and not idle because of more than one generation",
+			4,
+		},
+	}
+
+	expectedNotFullyCompacted := func(cp *tsm1.DefaultPlanner, reasonExp string, generationCountExp int64, testName string) {
+		compacted, reason := cp.FullyCompacted()
+		require.Equal(t, reason, reasonExp, "fullyCompacted reason", testName)
+		require.False(t, compacted, "is fully compacted", testName)
+
+		_, cgLen := cp.PlanLevel(1)
+		require.Zero(t, cgLen, "compaction group length; PlanLevel(1)", testName)
+		_, cgLen = cp.PlanLevel(2)
+		require.Zero(t, cgLen, "compaction group length; PlanLevel(2)", testName)
+		_, cgLen = cp.PlanLevel(3)
+		require.Zero(t, cgLen, "compaction group length; PlanLevel(3)", testName)
+
+		tsmP, pLenP := cp.Plan(time.Now().Add(-time.Second))
+		require.Zero(t, len(tsmP), "compaction group; Plan()", testName)
+		require.Zero(t, pLenP, "compaction group length; Plan()", testName)
+
+		_, cgLen, genLen := cp.PlanOptimize()
+		require.Equal(t, int64(1), cgLen, "compaction group length", testName)
+		require.Equal(t, generationCountExp, genLen, "generation count", testName)
+
+	}
+
+	for _, test := range furtherCompactedTests {
```

Comment:
Would [testing.T.Run](https://pkg.go.dev/testing#T.Run) be better here?  Sometimes it can complicate things, and if it doesn't help, there's no need to switch.

Interestingly, you can see a minimal example of `testing.T.Run` in the `testify /require` [documentation](https://pkg.go.dev/github.com/stretchr/testify/require#example-BoolAssertionFunc), or various places in our code.

Reply:
For example: [in the middleware](https://github.com/influxdata/influxdb/blob/7999835ac39fe9e197dd1c8024b1cd52cd3fca31/kit/transport/http/middleware_test.go#L13C1-L47C2)

I should have mentioned this in my last review.  Not a critical change, particularly if it complicates your test structure.  But, it gets rid of all the `testName` arguments to your `require.` methods (which may need formatting?).